internal/server/authn/errors.go

```diff
@@ -0,0 +1,13 @@
+package authn
+
+import (
+	"fmt"
+
+	"github.com/infrahq/infra/internal"
+)
+
+var (
+	ErrInvalidProviderURL          = fmt.Errorf("%w: %s", internal.ErrBadRequest, "invalid provider url")
+	ErrInvalidProviderClientID     = fmt.Errorf("%w: %s", internal.ErrBadRequest, "invalid provider client id")
+	ErrInvalidProviderClientSecret = fmt.Errorf("%w: %s", internal.ErrBadRequest, "invalid provider client secret")
```

Comment:
I'm not sure how well this pattern will scale. Usually I would want packages like `authn` to return their own errors, and then the api layer would translate those errors into the appropriate status code. That way all of our domain logic packages don't need to import the `internal` package.

That might look something like:

```go
var ErrValidation = fmt.Errorf("validation failed")

var ErrInvalidProviderURL = fmt.Errorf("%w: invalid provider url", ErrValidation)

```

And then `sendAPIError` would translate `authn.ErrValidation` to a 400 status code.

But I guess that's a big change from our current pattern, so not something we need to change right now. I think this approach works well (except for the very minor problem you encountered with repeated messages in the error string, which is solved well enough in this PR).

Reply:
Good suggestion, I changed to using a base `ErrValidation` in the authn package